### PR TITLE
DAOS-3821 test: pool update bad_evict yaml in pool create for CI regr…

### PR DESCRIPTION
**Pull request by dinghwah**
…ession failures

updated file: daos/src/tests/ftest/pool/bad_evict.py
updated file: daos/src/tests/ftest/pool/bad_query.py
updated file: daos/src/tests/ftest/pool/bad_evict.yaml
updated file: daos/src/tests/ftest/pool/bad_query.yaml

Test-tags: pr,-hw pool,badevict,badquery
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

### CODE REVIEW

File: src/tests/ftest/pool/bad_query.yaml [hunk 1]
```diff
@@ -29,3 +32,11 @@ querytests:
...
              - NULL
              - PASS
+   connectmode:
+     mode: 2
+   createmode:
+     mode: 511
+   createset:
+     setname: daos_server
+   createsize:
+     size: 1073741824
```
Comment (shimizukko) [inline src/tests/ftest/pool/bad_query.yaml]: Why did you add these? They are defined in L11-15 and passed in to `TestPool` when we call `self.pool.get_params(self)`

Comment (dinghwah) [reply src/tests/ftest/pool/bad_query.yaml]: bad_query.py is parsing and using the classical parameters.

Comment (shimizukko) [reply src/tests/ftest/pool/bad_query.yaml]: Where is it parsing?

Comment (phender) [reply src/tests/ftest/pool/bad_query.yaml]: @shimizukko is correct, these added yaml parameters are not used by the test.

Comment (dinghwah) [reply src/tests/ftest/pool/bad_query.yaml]: You are right, bad_query is using the new method in TestPool.
bad_evict is parsing the old parameters.

File: src/tests/ftest/pool/bad_evict.py [hunk 2]
```diff
@@ -140,5 +137,4 @@ def test_evict(self):
                         pool.uuid[item] = saveduuid[item]
                 if savedsvc is not None:
                     pool.svc = savedsvc
-                pool.disconnect()
-                pool.destroy(1)
+                pool.destroy(0)
```
Comment (phender) [inline src/tests/ftest/pool/bad_evict.py]: Given that a large majority of the pool tests were recently moved over to using `TestPool` to enable using dmg to create and destroy the pools, should this change be made here? or is a new PR required? @skirvan44

File: src/tests/ftest/pool/bad_evict.yaml [hunk 3]
```diff
@@ -22,7 +24,7 @@ evicttests:
      badname:
           setname:
              - NULL
-             - FAIL
+             - PASS
```
Comment (phender) [inline src/tests/ftest/pool/bad_evict.yaml]: These mux names should probably be updated to reflect the new expected behavior and add the missing valid name case:
```
   connectsetnames: !mux
      goodname:
         setname:
	    - daos_server
            - PASS
      wrongname:
         setname:
            - scott_server
            - PASS
      nullname:
         setname:
            - NULL
            - PASS
```

The code would suggest that only the current `badname` entry has an effect on the test.   Shouldn't the code in bad_evict.py look something more like this?
```
            # trash the pool group value
            savedgroup = pool.group
            if evictset is None:
                pool.group = None
            else:
                pool.group = evictset
```

Comment (dinghwah) [reply src/tests/ftest/pool/bad_evict.yaml]: Thanks for the feedbacks and comments, 
After some debug per your suggestion: 
pool.group is  <ctypes.c_char_Array_12 object at 0x7f5e311b2b90>
Current tests mux for pool.group with:
(1)valid pool.group (original object in pool.create for positive test)
(2)Null  (Null object for default pool.group test)

Comment (phender) [reply src/tests/ftest/pool/bad_evict.yaml]: Ah, yeah I forgot about the ctype conversion for DaosPool.group.  Updated code snippet:
```
            # trash the pool group value
            savedgroup = pool.group
            if evictset is None:
                pool.group = None
            else:
                pool.group.set_group(evictset)
```

Comment (dinghwah) [reply src/tests/ftest/pool/bad_evict.yaml]: resolved.
Please see the new commit#3.
